From: Michal Kubecek <mkubecek@suse.cz>
Date: Mon, 29 Apr 2019 07:33:12 +0200
Subject: rdma: fix argument of ZERO_PAGE() in rdma_umap_fault()
Patch-mainline: Not yet, tentative fix
References: http://lkml.kernel.org/r/20190429052136.GA21672@unicorn.suse.cz

rdma_umap_fault() passes vmf->vm_start to ZERO_PAGE() but vmf is a pointer
to struct vm_fault which does not have vm_start member. This doesn't matter
on x86_64 where ZERO_PAGE() doesn't actually use its argument but on some
other architectures, e.g. s390, it does and build fails.

Pass vmf->vma->vm_start instead as I believe that's what was intended.

Fixes: 67f269b37f9b ("RDMA/ucontext: Fix regression with disassociate")
Signed-off-by: Michal Kubecek <mkubecek@suse.cz>
---
 drivers/infiniband/core/uverbs_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/infiniband/core/uverbs_main.c
+++ b/drivers/infiniband/core/uverbs_main.c
@@ -895,7 +895,7 @@ static vm_fault_t rdma_umap_fault(struct vm_fault *vmf)
 
 	/* Read only pages can just use the system zero page. */
 	if (!(vmf->vma->vm_flags & (VM_WRITE | VM_MAYWRITE))) {
-		vmf->page = ZERO_PAGE(vmf->vm_start);
+		vmf->page = ZERO_PAGE(vmf->vma->vm_start);
 		get_page(vmf->page);
 		return 0;
 	}
